### ISSUE TITLE
state-timeline panel doesn't show data on initial load despite data being returned by datasource

### ISSUE THREAD

**Issue by WolfWFr** [grafana/grafana#48520]
**What happened**:
I open a grafana dashboard showing a state-timeline panel. The query behind it returns 2 datapoints, the timestamps of which lie before the start of the selected time window and right at the end of the selected timewindow, respectively. Directly after opening the dashboard, the panel does not show data. 

However, when refreshing the dashboard, or editing the panel, data suddenly appears and stays there, despite the call not returning any new datapoints.

![grafana empty state-timeline bug recording](https://user-images.githubusercontent.com/37142636/165932276-f49e9a73-6ec0-4e92-9353-da65239084ab.gif)

**What you expected to happen**:
I expect the data to become visible in the panel on the initial load, as the query behind it does already return data.

**How to reproduce it (as minimally and precisely as possible)**:
- create dashboard
- create state-timeline panel
- query data that returns datapoint prior to selected timeframe
- save
- reload webpage
- panel should be empty
- reload dashboard with grafana refresh button
- panel should suddenly show data

**Environment**: 
- Grafana version: v8.5.0-beta1 (behaviour was also observed in v8.5.0)
- Data source type & version: innius/grafana-simple-grpc-datasource v1.0.6
- OS Grafana is installed on: docker (grafana/grafana-enterprise)
- User OS & Browser: ubuntu 20.04.4 LTS; firefox 99.0 (64-bit)
- Grafana plugins: -
- Others: -

**Comment by zoltanbedi** [comment]
I was not able to reproduce this issue. Can you please give me a dashboard with the testDB datasource where you can reproduce this?

**Comment by grafanabot** [comment]
Thanks for creating this issue! We think it's missing some basic information. 

Follow the issue template and add additional information that will help us replicate the problem. 
For data visualization issues: 
- Query results from the inspect drawer (data tab & query inspector)
- Panel settings can be extracted in the panel inspect drawer JSON tab

For dashboard related issues: 
- Dashboard JSON can be found in the dashboard settings JSON model view

For authentication, provisioning and alerting issues, Grafana server logs are useful. 

Happy graphing!

**Comment by WolfWFr** [comment]
Hi zoltanbedi,

I have reviewed my earlier test and found that the two values being returned were wrongfully assigned to the same timestamp. After fixing this, the reported issue went away.

My sincerest apologies for taking your time.

The problem is now resolved for us. The reported issue still appears to occur when the panel initiates with two (identical) values with identical timestamps prior to the selected timeframe.

I will leave it to you to decide whether this behaviour is a bug or not. For us, however, the issue is resolved.
